install.sh

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env bash
+
+# Adapted from: https://github.com/rancher/k3d/blob/main/install.sh
+
+APP_NAME="fuseml-installer"
+REPO_URL="https://github.com/fuseml/fuseml"
+
+: "${USE_SUDO:=true}"
+: "${FUSEML_INSTALLER_INSTALL_DIR:=/usr/local/bin}"
+: "${VERIFY_CHECKSUM:=true}"
+
+HAS_OPENSSL="$(type "openssl" &> /dev/null && echo true || echo false)"
+
+# initArch discovers the architecture for this system.
+initArch() {
+  ARCH=$(uname -m)
+  case $ARCH in
+    armv5*) ARCH="armv5";;
+    armv6*) ARCH="armv6";;
+    armv7*) ARCH="arm";;
+    aarch64) ARCH="arm64";;
+    x86) ARCH="386";;
+    x86_64) ARCH="amd64";;
+    i686) ARCH="386";;
+    i386) ARCH="386";;
+  esac
+}
+
+# initOS discovers the operating system for this system.
+initOS() {
+  OS=$(uname|tr '[:upper:]' '[:lower:]')
+
+  case "$OS" in
+    # Minimalist GNU for Windows
+    mingw*) OS='windows'; USE_SUDO="false"; FUSEML_INSTALLER_INSTALL_DIR="/usr/bin"; EXTENSION=".exe";;
+  esac
+}
+
+# runs the given command as root (detects if we are root already)
+runAsRoot() {
+  if [ $EUID -ne 0 ] && [ $USE_SUDO = "true" ]; then
+    sudo "$@"
+  else
+    "$@"
+  fi
+}
+
+# verifySupported checks that the os/arch combination is supported for
+# binary builds.
+verifySupported() {
+  local supported="darwin-amd64\ndarwin-arm64\nlinux-amd64\nlinux-arm\nlinux-arm64\nwindows-amd64"
+  if ! echo "${supported}" | grep -q "${OS}-${ARCH}"; then
+    echo "No prebuilt binary for ${OS}-${ARCH}."
+    echo "To build from source, go to $REPO_URL"
+    exit 1
+  fi
+
+  if ! type "curl" > /dev/null && ! type "wget" > /dev/null; then
+    echo "Either curl or wget is required"
+    exit 1
+  fi
+
+  if [ "${VERIFY_CHECKSUM}" == "true" ] && [ "${HAS_OPENSSL}" != "true" ]; then
+    echo "In order to verify checksum, openssl must first be installed."
+    echo "Please install openssl or set VERIFY_CHECKSUM=false in your environment."
+    exit 1
+  fi
+}
+
+# checkFuseMLInstallerInstalledVersion checks which version of fusemnl-installer
+# is installed and if it needs to be changed.
+checkFuseMLInstallerInstalledVersion() {
+  if [[ -f "${FUSEML_INSTALLER_INSTALL_DIR}/${APP_NAME}" ]]; then
+    local version
+    version=$(fuseml-installer -v | cut -d" " -f3)
+    if [[ "$version" == "$TAG" ]]; then
+      echo "fuseml-installer ${version} is already ${DESIRED_VERSION:-latest}"
```

Comment:
the DESIRED_VERSION env variable doesn't seem to be used anywhere else

Reply:
fixed.